web-animations-2/Overview.bs

```diff
@@ -2353,6 +2376,152 @@ Items sorted earlier are executed before those sorted later.
+  ::
+    Then |state| is ''animation trigger state/idle'', and |did trigger| is set to false.
+
+  : Otherwise,
+  ::
+
+    <dl class=switch>
+      : If |trigger| is inside its [=animation trigger/active interval|active interval=],
+      ::
+        Then |state| is ''animation trigger state/primary'', and |did trigger| is set to true.
+
+      : Otherwise,
+      ::
+
+        <dl class=switch>
+          : If |did trigger| flag is false,
+          ::
+            Then |state| is ''animation trigger state/idle''.
+
+          : Otherwise,
+          ::
+            |state| is ''animation trigger state/inverse''.
+
+        </dl>
+
+    </dl>
+
+</dl>
+
+Issue: Should it be the correct/expected behavior for the "did trigger" flag to reset to false if the timeline becomes idle again?
+
+### Animation Trigger Types ### {#trigger-types}
+
+The behavior an [=animation trigger=] applies to its associated [=animation=]’s
+playback is defined by its <dfn export>animation trigger type</dfn>,
+which can be one of the following:
+
+<dl dfn-type=value dfn-for='animation trigger type'>
+  <dt><dfn>once</dfn>
+  <dd>
+    The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
+
+  <dt><dfn>repeat</dfn>
+  <dd>
+    The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
+    The ''inverse'' behavior is resetting the [=animation trigger state=] back to ''animation trigger state/idle''.
+
+  <dt><dfn>alternate</dfn>
+  <dd>
+    <dl class=switch>
+      : if the |did trigger| flag is false,
+      ::
+        The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
+
+      : Otherwise,
+      ::
+        The ''primary'' behavior is [=reverse an animation|reversing=] the associated animation.
+
+    </dl>
+
+    The ''inverse'' behavior is [=reverse an animation|reversing=] the associated animation.
+
+  <dt><dfn>state</dfn>
+  <dd>
+    The ''primary'' behavior is [=play an animation|triggering or resuming=] the associated animation.
+    The ''inverse'' behavior is [=pause an animation|pausing=] the associated animation.
+</dl>
+
+Issue: Need to bike-shed the name for type "state" type.
+
+### Animation Trigger Ranges ### {#trigger-ranges}
+
+An [=animation trigger=] has at least one [=animation attachment range|range=],
+its <dfn>default range</dfn>, which defines its [=animation trigger/active interval|active interval=].
+This range defines when the [=animation trigger|trigger=] is in its ''animation trigger state/primary'' state,
+as specified in [[#trigger-state]].
+
+An [=animation trigger=] may also have a second [=animation attachment range|range=],
+the <dfn>exit range</dfn>. The [=exit range=] is used to replace [=default range=] and extend
+the [=animation trigger/active interval|active interval=] of an [=animation trigger=] |trigger| as follows:
+
+<dl class=switch>
+  : If |trigger| is inside its [=animation trigger/active interval|active interval=],
```

Comment:
This should be based on the trigger state I think. Otherwise if you're in the exit range but not in the default range then this condition would be false.

Reply:
Right, refactored to use `|state|`